assets/js/util/i18n.js

```diff
@@ -33,28 +33,62 @@ import { __, sprintf } from '@wordpress/i18n';
  * For example, passing 65 returns '1m 5s'.
  *
  * @since 1.0.0
+ * @since n.e.x.t Refactored and renamed to improve localization.
+ * @private
  *
- * @param {number} seconds The number of seconds.
- * @return {string} Human readable string indicating time elapsed.
- *
+ * @param {number}                     seconds   The number of seconds.
+ * @param {(Intl.NumberFormatOptions)} [options] Optional formatting options.
+ * @return {string}   		Human readable string indicating time elapsed.
  */
-export const prepareSecondsForDisplay = ( seconds ) => {
+const durationFormat = ( seconds, options = {} ) => {
+	options = {
+		unitDisplay: 'short',
+		...options,
+		style: 'unit',
+	};
+
 	seconds = parseInt( seconds, 10 );
 
-	if ( isNaN( seconds ) || 0 === seconds ) {
-		return '0.0s';
+	if ( Number.isNaN( seconds ) ) {
+		seconds = 0;
+	}
+
+	let hours = Math.floor( seconds / 60 / 60 ) || '';
+	let minutes = Math.floor( ( seconds / 60 ) % 60 ) || '';
+
+	seconds = Math.floor( seconds % 60 );
+
+	if ( hours ) {
+		hours = numberFormat( hours, {
+			...options,
+			style: 'unit',
+			unit: 'hour',
+		} );
 	}
-	const results = {};
-	results.hours = Math.floor( seconds / 60 / 60 );
-	results.minutes = Math.floor( ( seconds / 60 ) % 60 );
-	results.seconds = Math.floor( seconds % 60 );
 
-	const returnString =
-		( results.hours ? results.hours + 'h ' : '' ) +
-		( results.minutes ? results.minutes + 'm ' : '' ) +
-		( results.seconds ? results.seconds + 's ' : '' );
+	if ( minutes ) {
+		minutes = numberFormat( minutes, {
+			...options,
+			style: 'unit',
+			unit: 'minute',
+		} );
+	}
 
-	return returnString.trim();
+	seconds = numberFormat( seconds, {
+		...options,
+		style: 'unit',
+		unit: 'second',
+	} );
+
+	const formattedString = sprintf(
+		/* translators: 1: seconds, 2: minutes, 3: hours */
+		_x( '%1$s %2$s %3$s', 'duration of time (numeric): hh mm ss', 'google-site-kit' ),
+		hours,
+		minutes,
+		seconds
```

Comment:
The resulting `formattedString` will be correct for the default locale here but `hours` should refer to `%3$s` where it is currently filling the `%1$s` placeholder. We just need to reverse the order of the placeholders in the string and the order they're passed to `sprintf` which should result in the same output but fix the relationship between placeholders and inputs. The translators comment is correct although it might be more clear to emphasize that each unit is formatted (rather than a bare number).

Reply:
Sorry, I think I misunderstood this last time round. Hopefully fixed now.